pipelines/test/ldd-check.yaml

```
@@ -54,6 +60,23 @@ pipeline:
         fi
         [ -z "$missing" ] && { pass "$f"; continue; }
         fail "$f: missing ${missing# }"
+      }
+      set -- $files
+      for f in "$@"; do
+        test_file "$f"
+      done
+      set -- $packages
+      for pkg in "$@"; do
+        echo "[ldd-check] Testing binaries in package $pkg"
+        apk info -eq "$pkg" > /dev/null || \
+          { fail "Package $pkg is not installed"; continue; }
+        apk info -Lq "$pkg" | while read f; do
+          [ -n "$f" ] || continue
+          f="/$f"
+          [ -f "$f" ] || continue
+          ldd "$f" > /dev/null 2>&1 || continue
```

This is weird, isn't it?
`ldd $file` fails, then we silently ignore that?

---

It is. I mentioned it in the original commit message (https://github.com/wolfi-dev/os/commit/8661705159195fe0da543412c78c3d6853fbcf41)

But I agree, it is confusing. Would this help?

```diff
diff --git a/pipelines/test/ldd-check.yaml b/pipelines/test/ldd-check.yaml
index a522428cb..2929e29da 100644
--- a/pipelines/test/ldd-check.yaml
+++ b/pipelines/test/ldd-check.yaml
@@ -62,6 +62,12 @@ pipeline:
         fail "$f: missing ${missing# }"
       }
 
+      is_dyn_obj() {
+        # ldd has this code already, let's not NIH it
+        # note: the actual ldd check happens in test_file()
+        ldd "$1" > /dev/null 2>&1
+      }
+
       test_files_in() {
         echo "[ldd-check] Testing binaries in package $pkg"
         apk info -eq "$pkg" > /dev/null || \
@@ -72,7 +78,7 @@ pipeline:
           [ -n "$f" ] || continue
           f="/$f"
           [ -f "$f" ] || continue
-          ldd "$f" > /dev/null 2>&1 || continue
+          is_dyn_obj "$f" || continue
           test_file "$f"
         done < "$tmpd/$pkg.list"
       }
```

---

https://github.com/wolfi-dev/os/pull/39042